### PR TITLE
Hide all finished tasks and show all tasks

### PR DESCRIPTION
**Pull request by nightism**
Two functions added:
1. Hide all finished deadline/floating tasks from the user interface.
2. Show all deadline/floating tasks after some of them are hidden from the user interface.

### CODE REVIEW

**Comment by nightism** [comment]
@pyokagan sorry for the late pull request

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -185,12 +190,16 @@ public void setDeadlineTaskFilter(Predicate<? super DeadlineTask> predicate) {
         filteredDeadlineTasks.setPredicate(predicate);
     }
 
+    @Override
+    public void setIsFinishedDeadlineFilter() {
+        setDeadlineTaskFilter(new DeadlineTaskFinishedPredicate());
+    }
```
Comment (pyokagan) [inline src/main/java/seedu/address/model/ModelManager.java]: @nightism Looks like you can just do `setDeadlineTaskFilter(new DeadlineTaskFinishedPredicate())` in `HideFinishedDeadlineCommand`, no?

Comment (pyokagan) [reply src/main/java/seedu/address/model/ModelManager.java]: Same goes for all the other methods you added to the `Model` interface just to set a filter.

**Review by pyokagan — CHANGES_REQUESTED** [review]
Avoid adding new methods to the `Model` interface unless absolutely necessary.

File: src/main/java/seedu/address/logic/commands/MarkDeadlineFinishedCommand.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ public CommandResult execute() {
                                                 true);
 
         try {
-            model.setDeadlineTask(targetIndex - 1, finishedDeadlineTask);
+            model.setDeadlineTask(targetIndex, finishedDeadlineTask);
```
Comment (pyokagan) [inline src/main/java/seedu/address/logic/commands/MarkDeadlineFinishedCommand.java:40]: @nightism Yeah, sorry about that...

File: src/main/java/seedu/address/model/ModelManager.java (hunk 1 above)
Comment (nightism) [reply src/main/java/seedu/address/model/ModelManager.java]: ok

**Review by pyokagan — CHANGES_REQUESTED** [review]
!!?!??!?!? The code does not even compile.

File: src/main/java/seedu/address/logic/commands/HideFinishedFloatingTaskCommand.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Hide all finished floating tasks from the user interface.\n"
+            + "Example: " + COMMAND_WORD + " f"; // TODO replace " f" after IndexPrefix.java added
+
+    public static final String MESSAGE_EDIT_TASK_SUCCESS = "All finished Floating Task hided.";
+
+    @Override
+    public CommandResult execute() {
+        model.setIsFinishedFloatingTaskFilter();
```
Comment (pyokagan) [inline src/main/java/seedu/address/logic/commands/HideFinishedFloatingTaskCommand.java]: Remove these kinds of methods and use `model.setFloatingTaskFilter()` instead.

**Comment by pyokagan** [comment]
Merged as d9ee9057c4cf961860fec3be42a59c0e613663cf

**Pull request closed without merging**